src/sql/parts/connection/common/connectionManagement.ts

```diff
@@ -253,6 +254,20 @@ export interface IConnectionManagementService {
 	 * Refresh the IntelliSense cache for the connection with the given URI
 	 */
 	rebuildIntelliSenseCache(uri: string): Thenable<void>;
+
+	/**
+	 * Get the current global connection, which is the connection from the active editor, unless OE
+	 * is focused or there is no such editor, in which case it comes from the OE selection. Returns
+	 * undefined when there is no such connection.
+	 * @returns {IConnectionProfile} The connection profile for the current global connection
+	*/
+	getCurrentGlobalConnection(): IConnectionProfile;
+
+	/**
+	 * Register the object explorer service so that the connection service can interact with it
+	 * @param service The object explorer service instance to register
+	 */
+	registerObjectExplorerService(service: IObjectExplorerService): void;
```

Comment:
is this a circular dependency?  maybe there is a some other more generic contract that OE can implement so that connection doesn't directly need an OE reference?